elementary/utils/dicts.py

```diff
@@ -0,0 +1,21 @@
+from typing import Dict, List
+
+
+# Flatten a nested dict by a given key, for example:
+# nested_dict = {"top1": "one", "nested": {"top1": "One", "top2": "Two"}}, flatten_by_key = "nested" -> {"top1": "One", "top2": "Two"}
+def flatten_dict_by_key(nested_dict: Dict, flatten_by_key: str) -> Dict:
+    flatten_dict = {**nested_dict, **nested_dict.get(flatten_by_key, {})}
+    flatten_dict.pop(flatten_by_key, None)
+    return flatten_dict
+
+
+# Merge dicts attributes into a single list that contains all the values.
+def merge_dicts_attribute(dicts: List[Dict], attribute_key: str) -> List:
```

Comment:
It will only work if the attributes are lists but we want to support attributes that are single item (like str, int dict) and merge them into a list as well 🙂 
So in order to make it work we will need to have 2 iterations to convert all to lists or have one big inline iteration with inline if statement inside it that uses the .get method on the same item 3 times:
```python
attributes = [
    d.get(attribute_key, []) if isinsatnce(d.get(attribute_key, []), list) else [d.get(attribute_key)]
    for d
    in dicts
]
return list(chain.from_iterable(attributes))
```

Reply:
no I checked, it works for both cases actually :)